### DISCUSSION TITLE
Record held MIDI notes that started before recording (especially in clip overdub)

### DISCUSSION THREAD

**Discussion by TubularCorpration** [helgoboss/helgobox#1268] [Ideas]
The way playtime works now, if I hold a note on my MIDI controller and then start recording a clip while continuing to hold the note, no MIDI data is recorded for the held note (but any other notes I play after I start recording are recorded as expected).  This also happens if I activate overdub on an existing clip, play a MIDI note and keep it held past when the clip loops - the note is recorded up to the end of the clip, but is cut off when the clip restarts.

In both cases I think it would be more intuitive if MIDI behaved more like audio by allowing you to record notes that sustain across the clip start point.  

When starting a new clip, that would mean checking for any incoming sustained notes and inserting new note-on messages for them at the start of the clip.  That would also work well enough for notes that sustain across the loop point while overdubbing - retriggering held notes wouldn't be perfect, but it would be better than cutting off the note altogether like it does now.

The main use case for me right now is improvising with an arpeggiator downstream from Playtime (in a hardware synth in this case, but a plugin arpeggiator would be the same) and recording the MIDI as-played into clips on the fly without releasing any notes I was already holding before I start recording, in a live setting where I have to be able to record phrases seamlessly without having notes drop out or needing to open up the clips in the MIDI editor to add back in notes that weren't recorded.

**Comment by helgoboss** [comment]
For normal recordings, this should actually work already. Provided you have at least a note-off. Do you have that?

For overdub it probably doesn't because I thought it's not necessary. Could maybe add that.

**Comment by TubularCorpration** [comment]
I did deliberately test it starting notes before I started recording into the clip and ending them before I stopped recording.  In that case (note on, then start recording, then note off, then stop recording) the note off message wasn't recorded at all.  If I double click the clip to open it in Reaper's MIDI editor, the last note I played during recording is highlighted in the piano roll but no MIDI data is recorded at all. If I play additional notes that both start and end while Playtime is recording, they are recorded correctly.

Let me double check and make absolutely sure I have the latest version of Reaper installed on my desktop.  I updated Playtime right before I started recording today but I didn't check what version of Reaper was installed and I haven't used that computer in a couple weeks.

**Comment by TubularCorpration** [comment]
I was running Reaper 7.24, but confirmed that the issue is still there with 7.25 (windows 10 x64), Playtime v2.16.10.

So to sum up
Note on -> start recording clip -> note off -> stop recording
gives me an empty clip, no note on or note off.

note on -> start recording clip -> play additional notes while sustaining first note - > notes off -> stop recording
gives me a clip containing only the notes that started after recording began.  The sustained note that started before recording began is absent (both note on and note off)

I guess this should be recategorized as a bug report?

**Comment by helgoboss** [comment]
Yes. That "normal recording doesn't pick up already playing notes" would be a bug, if it really doesn't work.

**Comment by TubularCorpration** [comment]
OK I created a new issue report for it and I'll close this thread.
